packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -122,7 +122,23 @@ abstract class FlutterCommand extends Command<Null> {
     _usesPubOption = true;
   }
 
-  void addBuildModeFlags({ bool defaultToRelease: true }) {
+  void usesBuildNumberOption() {
+    argParser.addOption('buildNumber',
+        help: 'An integer used as an internal version number.\n'
+            'On Android it is used as \'versionCode\'\n'
+            'On xcode builds it is used as \'CFBundleVersion\'',
+        valueHelp: '42');
```

Comment:
This would cause help text to show `--build-number=<42>`.  Instead, use `'int'`, so it'll show up as:

```
--build-number=<int>
```